src/Transport/Sending/MessageDispatcher.cs

```diff
@@ -128,6 +128,7 @@ async Task DispatchBatchForDestination(string destination, ServiceBusClient? cli
             var messageCount = messagesToSend.Count;
             int batchCount = 0;
             var sender = messageSenderRegistry.GetMessageSender(destination, client);
+            int maxItemsPerBatch = transaction == null ? int.MaxValue : 100;
```

Comment:
There is a magic limit of 4500 messages for non-transactional sends. That magic limit is depending on the tier and should be incorporated at some point into the `TryAdd` method. That's why I chose `int.Max` for now. 

https://github.com/Azure/azure-sdk-for-net/issues/21451

Happy to also set it to 4500 in the non-tx case if you prefer that.

Reply:
Realistically, this limit of 4500 probably cannot be reached with the amount of data that is added to a single message. Even if the body is empty, the headers are still big enough to make sure we never reach, 4500 msgs. I did a quick test with sending 4500 empty messages and it ended up batching roughly 250 into one single batch due to the size constraints.